authors/apps/authentication/serializers.py

```diff
@@ -10,10 +10,53 @@ class RegistrationSerializer(serializers.ModelSerializer):
 
     # Ensure passwords are at least 8 characters long, no longer than 128
     # characters, and can not be read by the client.
-    password = serializers.CharField(
-        max_length=128,
+    # serializers.RegexField - A text representation, that validates the given value matches against a certain regular expression.
+
+    password = serializers.RegexField(
+        regex="^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$",
         min_length=8,
-        write_only=True
+        max_length=128,
+        required=True,
+        write_only=True,
+        error_messages={
+            'required': 'Password is a required field',
+            'invalid': 'Password must have a number and a letter',
+            'min_length': 'Password must be at least 8 characters long',
+            'max_length': 'Password cannot be more than 128 characters'
+        }
+    )
+
+    # Email must be valid and unique
+    email = serializers.EmailField(
+        required=True,
+        validators=[
+            UniqueValidator(
+                queryset=User.objects.all(),
+                message='Email is already in use by another user'
+            )
+        ],
+        error_messages={
+            'invalid': 'Email must be of the format name@domain.com',
+            'required': 'Email is a required field'
+        }
+    )
+    # The username must be: unique, at least 5 characters long
+    # There should not be any space in the username
+    # Underscores and hyphens are also allowed in the username
+
```

Comment:
at least 5? I thought in the implementation you used 8..

Reply:
I think It should just be '# The username must be: unique'. Let me fix that. 
The password  is the one that must be at least 8 characters